src/test/java/book/store/onlinebookstore/book/BookServiceTest.java

```diff
@@ -0,0 +1,392 @@
+import book.store.onlinebookstore.exception.EntityNotFoundException;
+import book.store.onlinebookstore.mapper.BookMapper;
+import book.store.onlinebookstore.model.Book;
+import book.store.onlinebookstore.model.Category;
+import book.store.onlinebookstore.repository.book.BookRepository;
+import book.store.onlinebookstore.repository.book.BookSpecificationBuilder;
+import book.store.onlinebookstore.repository.category.CategoryRepository;
+import book.store.onlinebookstore.service.impl.BookServiceImpl;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @Mock
+    private BookSpecificationBuilder specificationBuilder;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Check if book is saved and correct dto is returned")
+    void save_ValidCreateBookRequestDto_ReturnsBookDtoWithId() {
+        //given
+        CreateBookRequestDto requestDto = new CreateBookRequestDto(
+                "testBook",
+                "testAuthor",
+                "123-123-0001",
+                BigDecimal.valueOf(100),
+                Set.of(1L, 2L),
+                null,
+                null
+        );
+
+        Book book = new Book();
+        book.setTitle(requestDto.title());
+        book.setAuthor(requestDto.author());
+        book.setIsbn(requestDto.isbn());
+        book.setPrice(requestDto.price());
+        Set<Category> categories = requestDto.categoriesIds().stream()
+                .map(id -> {
+                    Category category = new Category();
+                    category.setId(id);
+                    return category;
+                })
+                .collect(Collectors.toSet());
+        book.setCategories(categories);
+
+        BookDto expected = new BookDto();
+        expected.setId(1L);
+        expected.setTitle(requestDto.title());
+        expected.setAuthor(requestDto.author());
+        expected.setIsbn(requestDto.isbn());
+        expected.setPrice(requestDto.price());
+        expected.setCategoriesIds(requestDto.categoriesIds());
+
+        Mockito.when(bookMapper.toBook(requestDto)).thenReturn(book);

```

Comment:
```suggestion
        when(bookMapper.toBook(requestDto)).thenReturn(book);

```

Check all static imports